ghost/core/test/integration/services/mega.test.js

```diff
@@ -94,4 +101,117 @@ describe('MEGA', function () {
+
+            await fixtureManager.init('newsletters', 'members:newsletters');
+            await agent.loginAsOwner();
+            _sendEmailJob = require('../../../core/server/services/mega/mega')._sendEmailJob;
+            _mailgunClient = require('../../../core/server/services/bulk-email')._mailgunClient;
+        });
+
+        afterEach(function () {
+            mockManager.restore();
+        });
+
+        it('Tracks all the links in an email', async function () {
+            sinon.stub(_mailgunClient, 'getInstance').returns({});
+            const sendStub = sinon.stub(_mailgunClient, 'send');
+
+            sendStub.callsFake(async () => {
+                return {
+                    id: 'stubbed-email-id'
+                };
+            });
+
+            // Prepare a post and email model
+            const emailModel = await createPublishedPostEmail();
+
+            // Launch email job
+            await _sendEmailJob({emailModel, options: {}});
+
+            await emailModel.refresh();
+            emailModel.get('status').should.eql('submitted');
+
+            // Get email data (first argument to sendStub call)
+            const emailData = sendStub.args[0][0];
+            const recipientData = sendStub.args[0][1];
+            const replacements = sendStub.args[0][2];
+
+            const recipient = recipientData[Object.keys(recipientData)[0]];
+
+            // Do the actual replacements for the first member, so we don't have to worry about them anymore
+            replacements.forEach((replacement) => {
+                emailData[replacement.format] = emailData[replacement.format].replace(
+                    replacement.match,
+                    recipient[replacement.id]
+                );
+
+                // Also force Mailgun format
+                emailData[replacement.format] = emailData[replacement.format].replace(
+                    `%recipient.${replacement.id}%`,
+                    recipient[replacement.id]
+                );
+            });
+
+            const html = emailData.html;
+
+            const memberUuid = recipient.unique_id;
+
+            // Test if all links are replaced and contain the member id
+            const cheerio = require('cheerio');
+            const $ = cheerio.load(html);
+
+            const exclude = '%recipient.unsubscribe_url%';
+            let firstLink;
+    
+            for (const el of $('a').toArray()) {
+                const href = $(el).attr('href');
+
+                if (href === exclude) {
+                    continue;
+                }
+
+                // Check if the link is a tracked link
+                assert(href.includes('?m=' + memberUuid), href + ' is not tracked');
+
+                if (!firstLink) {
+                    firstLink = new URL(href);
+                }
+            }
+            const links = (await models.LinkRedirect.findAll({post_id: emailModel.get('post_id')})).models;
```

Comment:
Would be cool if we can use the repository or the API for this - otherwise gonna have to update this as part of the DB rename.